demos/guide/quic-hq-interop-server.c

```diff
@@ -0,0 +1,680 @@
+ * QUIC streams. Each connection is handled until termination, and streams are
+ * processed individually using the `process_new_stream` function.
+ *
+ * @param ctx Pointer to the SSL_CTX object configured for QUIC.
+ * @param sock  BIO of the bound UDP socket.
+ *
+ * @return Returns 0 on error; otherwise, the server runs indefinitely.
+ *
+ * Operation:
+ * - Creates a QUIC listener using the provided SSL_CTX and associates it
+ *   with the specified UDP socket.
+ * - Waits for incoming QUIC connections and accepts them.
+ * - For each connection:
+ *   - Accepts incoming streams.
+ *   - Processes each stream using `process_new_stream`.
+ *   - Shuts down the connection upon completion.
+ *
+ * Error Handling:
+ * - If listener creation or connection acceptance fails, the function logs
+ *   an error message and exits the loop.
+ * - Cleans up allocated resources (e.g., listener, connection) on failure.
+ *
+ * Usage:
+ * - Call this function in the main server loop after setting up the
+ *   SSL_CTX and binding a UDP socket.
+ *
+ * Notes:
+ * - Uses blocking operations for listener, connection, and stream handling.
+ * - Incoming streams are processed based on the configured stream policy.
+ * - The server runs in an infinite loop unless a fatal error occurs.
+ */
+static int run_quic_server(SSL_CTX *ctx, BIO *sock)
+{
+    int ok = 0;
+    SSL *listener, *conn, *stream;
+    unsigned long errcode;
+
+    /*
+     * Create a new QUIC listener. Listeners, and other QUIC objects, default
+     * to operating in blocking mode. The configured behaviour is inherited by
+     * child objects.
+     */
+    if ((listener = SSL_new_listener(ctx, 0)) == NULL)
+        goto err;
+
+    /* Provide the listener with our UDP socket. */
+    SSL_set_bio(listener, sock, sock);
+
+    /* Begin listening. */
+    if (!SSL_listen(listener))
+        goto err;
+
+    /*
+     * Begin an infinite loop of listening for connections. We will only
+     * exit this loop if we encounter an error.
+     */
+    for (;;) {
+        /* Pristine error stack for each new connection */
+        ERR_clear_error();
+
+        /* Block while waiting for a client connection */
+        printf("Waiting for connection\n");
+        conn = SSL_accept_connection(listener, 0);
+        if (conn == NULL) {
+            fprintf(stderr, "error while accepting connection\n");
+            goto err;
+        }
+        printf("Accepted new connection\n");
+
+        if (!SSL_set_incoming_stream_policy(conn,
+                                            SSL_INCOMING_STREAM_POLICY_ACCEPT,
+                                            0)) {
+            fprintf(stderr, "Failed to set incomming stream policy\n");
+            goto err;
+        }
+
+        /*
+         * Until the connection is closed, accept incomming stream
+         * requests and serve them
+         */
+        for (;;) {
+            stream = SSL_accept_stream(conn, 0);
```

Comment:
We should also repeat here the `SSL_accept_stream()` blocks.

Reply:
Can you elaborate please?  I'm not quite sure what you are suggesting here.

Reply:
I meant to say add a comment the SSL_accept_stream() is blocking.

Reply:
added